### ISSUE TITLE
Containers can't be built on non-nixos system (debian)

### ISSUE THREAD

**Issue by Melkor333** [erikarvstedt/extra-container#22]
I'm trying to run NixOS containers on a Debian host but I get the following error:

```
error: attribute 'kernelPackages' missing, at /nix/var/nix/profiles/per-user/root/channels/nixpkgs/nixos/modules/virtualisation/nixos-containers.nix:283:19
```

Can this be fixed somehow or is the tool only usable on NixOS systems? (Since the README contains install instructions for "other systemd-based systems" I assumed that is the case :) ).

**Comment by boxofrox** [comment]
I ran into the 'kernelPackages' error on NixOS 21.05.  Perhaps the recent versions of nixpkgs are not compatible with extra-container.

I'm honestly scratching my head as to why a container needs a kernel.

**Comment by Patryk27** [comment]
@boxofrox it's related to https://github.com/erikarvstedt/extra-container/pull/18 (with the rationale tl;dr at https://github.com/NixOS/nixpkgs/pull/100433)

**Comment by erikarvstedt** [comment]
Fixed in the latest release (0.7).
